tests/unit/test_server.py

```diff
@@ -40,28 +40,121 @@ class TestInstall(BaseUnitCase):
 
     def setUp(self):
         self.remove_runs_once_flag(server.status)
+        self.remove_runs_once_flag(server.install)
         self.maxDiff = None
         super(TestInstall, self).setUp(capture_output=True)
 
+    @patch('prestoadmin.server._find_local_presto_rpm')
     @patch('prestoadmin.server.package.check_if_valid_rpm')
     @patch('prestoadmin.server.execute')
-    def test_install_server(self, mock_execute, mock_check_rpm):
-        local_path = "/any/path/rpm"
-        server.install(local_path)
-        mock_check_rpm.assert_called_with(local_path)
+    def test_install_server_local_path(self, mock_execute, mock_check_rpm,
+                                       mock_find_local_rpm):
+        rpm_specifier = "/any/path/rpm"
+        mock_find_local_rpm.return_value = rpm_specifier
+        env.rpm_source = 'facebook'
+        server.install(rpm_specifier)
+        mock_check_rpm.assert_called_with(rpm_specifier)
         mock_execute.assert_called_with(server.deploy_install_configure,
-                                        local_path, hosts=get_host_list())
+                                        rpm_specifier, hosts=get_host_list())
+
+    @patch('prestoadmin.server._find_local_presto_rpm')
```

Comment:
Here and throughout, this is kind of a weird use of mock objects. Generally, you want to try to mock calls to things that don't exist in testing (some prod-only resource), are expensive to call (downloading the world), or are hard to force into a specific error state (return a 451 error from a web server).

The way this is mocked is pretty tightly coupled with the current implementation, which adds a lot of maintenance cost. A more general approach might be along the following lines:

0. Mock os.path.isfile so that find_local_presto_rpm returns the value you pass into server.install
1. Mock package.local so that check_if_valid_rpm thinks the rpm is good.
2. Mock urllib2.urlopen and assert we don't call it.
3. Leave the execute mock as it is.